### PR TITLE
New Serverless Pattern: s3 - lambda - bedrock - cdk python

### PR DESCRIPTION
**Pull request by pdineshbalaaji**
…ttern

*Issue #, if available:*

*Description of changes:*
This pattern creates a Amazon S3 bucket, a Lambda function, which will be used to get vector embeddings for the document uploaded into Amazon S3 using Amazon Bedrock.
The Lambda function is triggered by S3 'OBJECT_CREATED' events. The Lambda function can optionally apply business logic to the S3 object. After processing, the Lambda function invokes the embedding model form Amazon Bedrock to get the vector embeddings, which is useful for LLM RAG use cases.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: s3-lambda-bedrock-cdk-python/.gitignore
Comment (parikhudit) [inline s3-lambda-bedrock-cdk-python/.gitignore:1]: This file is not needed

File: s3-lambda-bedrock-cdk-python/tests/unit/test_s3_lambda_bedrock_cdk_python_stack.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from s3_lambda_bedrock_cdk_python.s3_lambda_bedrock_cdk_python_stack import S3LambdaBedrockCdkPythonStack
+
+# example tests. To run these tests, uncomment this file along with the example
+# resource in s3_lambda_bedrock_cdk_python/s3_lambda_bedrock_cdk_python_stack.py
+def test_sqs_queue_created():
+    app = core.App()
+    stack = S3LambdaBedrockCdkPythonStack(app, "s3-lambda-bedrock-cdk-python")
+    template = assertions.Template.from_stack(stack)
+
+#     template.has_resource_properties("AWS::SQS::Queue", {
```
Comment (parikhudit) [inline s3-lambda-bedrock-cdk-python/tests/unit/test_s3_lambda_bedrock_cdk_python_stack.py:13]: Extra code - please remove

**Comment by parikhudit** [comment]
Thanks for the contribution @pdineshbalaaji. Good solution, however there's a similar pattern available at https://serverlessland.com/patterns/s3-lambda-bedrock. It is recommended to check https://serverlessland.com/patterns for existing patterns before creating a PR to save some time. For now, I'm closing this PR, however if you can always create a new PR if would to update the existing pattern or submit a new & unique pattern. Thanks again.

**Comment by pdineshbalaaji** [comment]
@parikhudit I verified the existing pattern and it seems like its a SAM pattern tagged incorrectly as CDK pattern.
This one I have submitted is a CDK pattern. Can we re-open this pattern for consideration and correct the existing pattern in the reporisoty?
Let me know what would be the next step?

**Comment by parikhudit** [comment]
> @parikhudit I verified the existing pattern and it seems like its a SAM pattern tagged incorrectly as CDK pattern. This one I have submitted is a CDK pattern. Can we re-open this pattern for consideration and correct the existing pattern in the reporisoty? Let me know what would be the next step?

@pdineshbalaaji thanks for the heads up. We'd check and update the framework type on that PR. Please check the review comments added earlier and remove extra files, remove extra lines of code that's not needed and test it once as per the steps provided in README.

Once all good, happy to review and merge it.

File: s3-lambda-bedrock-cdk-python/tests/unit/test_s3_lambda_bedrock_cdk_python_stack.py (hunk 1 above)
Comment (pdineshbalaaji) [reply s3-lambda-bedrock-cdk-python/tests/unit/test_s3_lambda_bedrock_cdk_python_stack.py]: extra lines of code removed

**Comment by pdineshbalaaji** [comment]
@parikhudit Thank you! I have resolved the review comments and tested it to make sure its working as expected.
image

**Comment by parikhudit** [comment]
Thanks @pdineshbalaaji for making additional changes as suggested. LGTM. DA team would merge the code and publish the pattern.

**Pull request merged**